Fails to build on darwin with:

```
building
build flags: SHELL=/nix/store/k89nm2jva0qmvd970f84wq2iq1iwm9bs-bash-4.4-p23/bin/bash PREFIX=\$\(out\)/bin
scripts/make.sh
Generate headers from toys/*/*.c...
generated/newtoys.h Library probe...........
Make generated/config.h from .config.
generated/flags.h generated/globals.h generated/tags.h generated/help.h
Compile toybox...............lib/portability.c:474:10: warning: unused variable 'offset' [-Wunused-variable]
  int i, offset;
         ^
1 warning generated.
.............................................toys/posix/cp.c:329:30: error: implicit declaration of function 'utimensat' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
      if (fdout == AT_FDCWD) utimensat(cfd, catch, times, AT_SYMLINK_NOFOLLOW);
                             ^
toys/posix/cp.c:329:30: note: did you mean 'utimes'?
/nix/store/i09q4si3r6ad3iblrdjxfmvl8bks6amg-Libsystem-osx-10.12.6/include/sys/time.h:202:5: note: 'utimes' declared here
int     utimes(const char *, const struct timeval *);
        ^
toys/posix/cp.c:330:12: error: implicit declaration of function 'futimens' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
      else futimens(fdout, times);
           ^
toys/posix/cp.c:330:12: note: did you mean 'futimes'?
/nix/store/i09q4si3r6ad3iblrdjxfmvl8bks6amg-Libsystem-osx-10.12.6/include/sys/time.h:190:5: note: 'futimes' declared here
int     futimes(int, const struct timeval *);
        ^
2 errors generated.
toys/posix/cpio.c:215:15: error: implicit declaration of function 'utimensat' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        err = utimensat(AT_FDCWD, name, times, AT_SYMLINK_NOFOLLOW);
              ^
toys/posix/cpio.c:215:15: note: did you mean 'utimes'?
/nix/store/i09q4si3r6ad3iblrdjxfmvl8bks6amg-Libsystem-osx-10.12.6/include/sys/time.h:202:5: note: 'utimes' declared here
int     utimes(const char *, const struct timeval *);
        ^
1 error generated.
..make: *** [Makefile:17: toybox] Error 1
```

--- a/pkgs/tools/misc/toybox/default.nix
+++ b/pkgs/tools/misc/toybox/default.nix
@@ -7,13 +7,13 @@
 
 stdenv.mkDerivation rec {
   pname = "toybox";
-  version = "0.8.3";
+  version = "0.8.4";
 
   src = fetchFromGitHub {
     owner = "landley";
     repo = pname;
     rev = version;
-    sha256 = "0cb1n0skanwwkwgzlswwhvfb4iji1bw9iqskmczlhakpw3j1yaqa";
+    sha256 = "0cgbmv6qk1haj709hjx5q4sl7wgh91i459gzs1203adwc7rvk6jv";
   };
 
   buildInputs = lib.optionals enableStatic [ stdenv.cc.libc stdenv.cc.libc.static ];
@@ -60,6 +60,8 @@ stdenv.mkDerivation rec {
     homepage = "https://landley.net/toybox/";
     license = licenses.bsd0;
     platforms = with platforms; linux ++ darwin ++ freebsd;
+    # https://github.com/NixOS/nixpkgs/issues/101229
+    broken = stdenv.isDarwin;
     maintainers = with maintainers; [ hhm ];
     priority = 10;
   };

